src/hyperspectral/hyperspectral/target/matched_filter.py

```diff
@@ -0,0 +1,15 @@
+import math
+
+import numpy as np
+
+from hyperspectral.math.rpca import whitening_matrix, whiten
+
+def normalized_matched_filter(image, target, clutter_cov):
+    """
+
+    """
+    white = whitening_matrix(clutter_cov)
+    X̃ = whiten(image, white_matrix=white)
+    s̃ = whiten(target, white_matrix=white)
+
+    return np.divide(np.einsum('i,rci->rc', s̃, X̃), np.sqrt(np.einsum('rci,rci->rc',X̃, X̃)))/math.sqrt(np.inner(s̃, s̃))
```

Comment:
I understand the denominator, but I need some help decrypting the numerator.

Reply:
For every element of the whitened image, we get a dot product of the whitened signal and the image pixel spectrum.  And elementwise we divide that dot product with the 2-norm of the whitened image pixel.  This all boils down to `cos(θ)` where θ is the angle formed by the whitened pixel vector and the whitened signal vector.

Reply:
Okay, sounds reasonable, I will take a second look at the code to understand exactly tomorrow.